Fabric.Identity.API/scripts/Install-Identity.ps1

```diff
@@ -71,14 +68,40 @@ $currentDirectory = $PSScriptRoot
-$selectedCerts = Get-Certificates -primarySigningCertificateThumbprint $installSettings.primarySigningCertificateThumbprint -encryptionCertificateThumbprint $installSettings.encryptionCertificateThumbprint -installConfigPath $configStore.Path -scope $installSettingsScope -quiet $quiet
-$iisUser = Get-IISAppPoolUser -credential $credential -appName $installSettings.appName -storedIisUser $installSettings.iisUser -installConfigPath $configStore.Path -scope $installSettingsScope
-Add-PermissionToPrivateKey $iisUser.UserName $selectedCerts.SigningCertificate read
 $appInsightsKey = Get-AppInsightsKey -appInsightsInstrumentationKey $installSettings.appInsightsInstrumentationKey -installConfigPath $configStore.Path -scope $installSettingsScope -quiet $quiet
 $sqlServerAddress = Get-SqlServerAddress -sqlServerAddress $commonInstallSettings.sqlServerAddress -installConfigPath $configStore.Path -quiet $quiet
 $identityDatabase = Get-IdentityDatabaseConnectionString -identityDbName $installSettings.identityDbName -sqlServerAddress $sqlServerAddress -installConfigPath $configStore.Path -quiet $quiet
 $metadataDatabase = Get-MetadataDatabaseConnectionString -metadataDbName $commonInstallSettings.metadataDbName -sqlServerAddress $sqlServerAddress -installConfigPath $configStore.Path -quiet $quiet
 
+# Switch to DosInstallUtilities later, testing with Fabric-Install-Utilities locally
+try {
+    $encryptionCertificate = Get-Certificate -certificateThumbprint $commonInstallSettings.encryptionCertificateThumbprint
+    $fabricInstallerSecret = Get-DecryptedString -encryptionCertificate $encryptionCertificate -encryptedString $commonInstallSettings.fabricInstallerSecret
+}
+catch {
+    # create new cert/secret if anything bad happens
+    Remove-IdentityEncryptionCertificate -encryptionCertificateThumbprint $commonInstallSettings.encryptionCertificateThumbprint
+    $encryptionCertificate = New-IdentityEncryptionCertificate
+    $fabricInstallerSecret = Invoke-ResetFabricInstallerSecret -identityDbConnectionString $identityDatabase.DbConnectionString
+}
+
+# Create new cert if current is expired/expiring soon
+$certIsValid = Test-IdentityEncryptionCertificateValid -encryptionCertificate $encryptionCertificate
+if ($certIsValid -eq $false) {
+    Remove-IdentityEncryptionCertificate -encryptionCertificateThumbprint $commonInstallSettings.encryptionCertificateThumbprint
+    $encryptionCertificate = New-IdentityEncryptionCertificate
+}
+
+# Wait until potentially creating new certificate to add permissions to private key
+$iisUser = Get-IISAppPoolUser -credential $credential -appName $installSettings.appName -storedIisUser $installSettings.iisUser -installConfigPath $configStore.Path -scope $installSettingsScope
+Add-PermissionToPrivateKey $iisUser.UserName $encryptionCertificate read
+
+# update install.config
+# Assumes both encryption and signing cert are the same certificate
```

Comment:
Important note here ⏫

Reply:
Currently, the only way to separate the encryption/signing certificate is to do so manually through the install.config.  
The Get-Certificates behavior assumed that these would be the same as well.
With this new implementation, everytime the new cert is generated, then both the encryption/signing certs are replaced.

Is this assumption still valid?

Reply:
I always remembered them showing the same value, and it seemed weird to me that they are the same but separated out.